### ISSUE TITLE
Dump operator to enable easier debugging 

### ISSUE THREAD

**Issue by pditommaso** [nextflow-io/nextflow#517]
The `dump` operator has a semantic similar to the `view` operator, however it output is printed to the system logging instead of the standard output. 

The channel output is dumped when the command line option `-dump-channel` is specified. An option list of tags can be provide to enable only those operators specifying a matching `tag` option.

**Comment by baffelli** [comment]
Would love to see this operator! At the moment I'm using a workaround with map:
```
chan.map{
it->
outFile = new File("${workflow.workDir}/${workflow.runName}.log")
outFile.text = it.join("\n")
return it
}
```

**Comment by pditommaso** [comment]
The `tag` information could be used as a custom [Marker](https://www.slf4j.org/faq.html#fatal) to better control the log output.

**Comment by pditommaso** [comment]
Available for test in version `0.27.0-SNAPSHOT`.
